docs/modules/ROOT/pages/index.adoc

```diff
@@ -0,0 +1,175 @@
+:pyenv: link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.4.0b/src/openzeppelin/access/ownable/library.cairo[pyenv]
+
+= Overview
+
+Nile is a CLI tool to develop or interact with StarkNet projects written in Cairo. It consists of different components for developing, compiling, testing, and deploying your smart contracts and dApps, providing a CLI for executing tasks, and a Runtime Environment (NRE) for scripting.
+
+The package is designed to be extensible and very customizable by using plugins. This guide will take you through the installation of our recommended setup, but as we expect a lot of functionality coming from plugins, you are free to customize it.
+
+TIP: You can get a list of plugins extending Nile functionality from the xref:plugins.adoc[Plugins] section.
+
+:oz-account: link:https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/account/presets/Account.cairo[Openzeppelin Account]
+
+== Quickstart
+
+Using a sample contract, we will explore the basics of creating a Nile project. We will test it locally, deploy an account and the contract to a devnet node, and send transactions through the account to the contract.
+
+=== Requirements
+
+==== GMP for fastecdsa
+
+Before installing Cairo on your machine, you need to install `gmp`:
+
+[,bash]
+----
+sudo apt install -y libgmp3-dev # linux
+brew install gmp # mac
+----
+
+TIP: If you have any trouble installing gmp on your Apple M1 computer, https://github.com/OpenZeppelin/nile/issues/22[here's a list of potential solutions].
+
+==== Supported Python versions
+
+Some Nile dependencies have specific python version requirements. Therefore we recommend using a python version manager like {pyenv}, and virtual environments to avoid conflicts.
+
+IMPORTANT: Currently supported Python versions are >=3.8 and <3.10.
+
+=== Installation
+
+Create a folder for your project and cd into it:
+
+[,sh]
+----
+mkdir myproject && cd myproject
+----
+
+Create a virtualenv and activate it:
+
+[,sh]
+----
+python3 -m venv env
+source env/bin/activate
+----
+
+IMPORTANT: Review xref:index.adoc#requirements[Requirements]
+
+Install Nile:
+
+[,sh]
+----
+(env): pip install cairo-nile
```

Comment:
```suggestion
pip install cairo-nile
```
not sure about adding shell prefixes to these instructions

Reply:
I think it could help the users to avoid making the mistake of not being inside the env. I also think it could add clarity to add a dollar $ sign before commands to differentiate it from the output, like:

<img width="789" alt="Screenshot 2022-12-27 at 17 13 39" src="https://user-images.githubusercontent.com/26070224/209692555-d45e4ad4-a319-4d86-a5ea-e651761fe414.png">

Reply:
i think the `$` or `(env)` should only be there when we're simply showing how it looks. when it comes to instructions/steps to follow, i favor the plain command because it's easier to copy and paste. the `$` hinders the experience of using this button:

<img width="54" alt="image" src="https://user-images.githubusercontent.com/1703831/209766655-f7ae421f-58d9-468c-b213-2709df4190e2.png">

Reply:
That's a great point. Following up on that, should we remove then the output from the code boxes? And what about the code boxes with multiple commands or examples for mac and linux?

```
(env): nile compile
🤖 Compiling all Cairo contracts in the contracts directory
🔨 Compiling contracts/contract.cairo
✅ Done
```

```
sudo apt install -y libgmp3-dev # linux
brew install gmp # mac
```

```
python3 -m venv env
source env/bin/activate
```

Reply:
multiple command ones should work as they are. the mac and linux is one an acceptable exception, but the answer for all cases is that it they can be split up:

```sh
nile compile
```
```sh
🤖 Compiling all Cairo contracts in the contracts directory
🔨 Compiling contracts/contract.cairo
✅ Done
```
------
```sh
sudo apt install -y libgmp3-dev # linux
```
```sh
brew install gmp # mac
```